File: django_graphql/sales/graphql/queries/sales_query.py
import graphene
from sales.graphql.types.salestype import SalesModelType
from sales.models import Sales

class SalesQuery(graphene.ObjectType):
    sales_list = graphene.List(SalesModelType)

    def resolve_sales_list(root, info):                     
        return Sales.objects.all()


# =====REQUEST======
# query GetSales {
#   getSales{
#     salesList {
#       id
#       salesamount
#       salesdate
#     }
#   }
# }

File: vite-react/src/graphql/sales_query.ts
import { gql } from '@apollo/client';

export const SALES_QUERY = gql`
  query GetSales {
    getSales{
      salesList {
        salesamount
        salesdate
      }
    }
  }
`;


export interface SaleData {
    salesamount: number
    salesdate: string | number
}


export interface SalesListData {
  getSales: {
    salesList: SaleData[]
  }
}






